spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcOpenRowSetTemplate.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jdbc.repository.support;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.springframework.jdbc.InvalidResultSetAccessException;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.datasource.DataSourceUtils;
+import org.springframework.jdbc.support.JdbcUtils;
+import org.springframework.util.Assert;
+
+/**
+ * An extended {@link NamedParameterJdbcTemplate} to return an {@link JdbcOpenSqlRowSet}.
+ *
+ * @author detinho
+ */
+class JdbcOpenRowSetTemplate extends NamedParameterJdbcTemplate {
+
+    JdbcOpenRowSetTemplate(JdbcOperations classicJdbcTemplate) {
+        super(classicJdbcTemplate);
+    }
+
+    JdbcOpenSqlRowSet queryForOpenCursorRowSet(String sql, SqlParameterSource paramSource, Integer fetchSize)  {
```

Comment:
I'm not sure why this builds on `RowSet` and not on `ResultSet`.

Reply:
I wrapped a reference to the ResultSet and the corresponding DataSource so that it can be released later, when the stream is closed. I used the RowSet interface as this code can be used as a base for further work just as in the issue you linked above.